toolchain/check/check.cpp

```diff
@@ -212,6 +212,56 @@ struct UnitInfo {
   UnitInfo* api_for_impl = nullptr;
 };
 
+// Collects transitive imports, handling deduplication.
+static auto CollectTransitiveImports(const UnitInfo::PackageImports& imports,
+                                     int total_ir_count)
+    -> llvm::SmallVector<SemIR::ImportIR> {
+  llvm::SmallVector<SemIR::ImportIR> results;
+
+  // Track whether an IR was imported in full, including `export import`. This
+  // distinguishes from IRs that are indirectly added without all names being
+  // exported to this IR.
+  llvm::SmallVector<int> imported_irs(total_ir_count, -1);
```

Comment:
Perhaps `imported_ir_indexes`? It would have been useful to me for the comment to mention this is tracking indexes in `results`.

Reply:
How about `ir_to_result_index`